onnxruntime/core/graph/data_propagation/gather_op_data_propagation.cc

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "gather_op_data_propagation.h"
+#include "core/common/common.h"
+#include "core/graph/node_arg.h"
+#include "core/graph/onnx_protobuf.h"
+#include "core/providers/common.h"
+
+namespace onnxruntime {
+
+Status GatherOpDataPropagation::infer() {
+  int dim_size = 0;
+  if (output_from_onnx_op_data_propagation_.has_tensor_type() &&
+      output_from_onnx_op_data_propagation_.tensor_type().has_shape()) {
+    dim_size = output_from_onnx_op_data_propagation_.tensor_type().shape().dim_size();
+  }
+  ORT_ENFORCE(dim_size == 0);
```

Comment:
Not clear here, comments would help

Reply:
This throws. Should this return an error code? In any case, the error message would help.
Also, I understand why I raised this question in the first place. The var is declared as if it is going to be used later. But it is not necessary. The condition can be checked inside the above if and error is raised if necessary.